pkg/metrics/latency_reporter.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"knative.dev/pkg/metrics"
+)
+
+const LatencyMetricName string = "process_latencies"
+
+type LatencyReporter struct {
+	durationInMsecM *stats.Float64Measure
+}
+
+type ProcessType string
+
+const (
+	ResourceUpdateToBrokerCellNotified ProcessType = "ResourceUpdateToBrokerCellNotified"
+)
+
+func (r *LatencyReporter) register() error {
+	return metrics.RegisterResourceView(
+		&view.View{
+			Name:        r.durationInMsecM.Name(),
+			Description: r.durationInMsecM.Description(),
+			Measure:     r.durationInMsecM,
+			Aggregation: view.Distribution(metrics.Buckets125(1, 100000)...), // 1, 2, 5, 10, 20, 50, 100, 1000, 5000, 10000, 20000, 50000, 1000000
+			TagKeys: []tag.Key{
+				NamespaceNameKey,
+				ProcessTypeKey,
+				EntityNameKey,
```

Comment:
I would also add the resource name as a key to make the time series more unique. Stackdriver has a limit of only accepting a unique timeseries every 10s: https://cloud.google.com/monitoring/quotas#custom_metrics_quotas

Reply:
Good to know, accounted for it.